gtclang/test/atlas-integration-test/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+##===------------------------------------------------------------------------------*- CMake -*-===##
+##                          _                      
+##                         | |                     
+##                       __| | __ ___      ___ ___  
+##                      / _` |/ _` \ \ /\ / / '_  | 
+##                     | (_| | (_| |\ V  V /| | | |
+##                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+##
+##
+##  This file is distributed under the MIT License (MIT). 
+##  See LICENSE.txt for details.
+##
+##===------------------------------------------------------------------------------------------===##
+
+if(eckit_DIR AND atlas_DIR)
+
+    find_package(eckit)
+    if(NOT eckit_FOUND)
+            message(FATAL_ERROR "Could not find eckit")
+    endif()
+
+    find_package(atlas)
+    if(NOT atlas_FOUND)
+            message(FATAL_ERROR "Could not find atlas")
+    endif()
+
+    # Need to specify here the names of the stencil codes that are going to be generated.
+    set(generated_stencil_codes generated_copyCell.hpp generated_copyEdge.hpp generated_accumulateEdgeToCell.hpp generated_diffusion.hpp)
```

Comment:
for now the convention was to have them be named *.cpp even though we include them

Reply:
will change

Reply:
There's a problem: if they are cpp files, the compiler is going to try to also compile them as separate translation units (and it will fail as other includes are missing), because they are listed as sources of build target `AtlasIntegrationTestCompareOutput` in order to make the target dependent on them.

Reply:
@twicki After having discussed this with you, I tried to do the same thing as gtclang `integration-test`'s cmake does which is adding an additional target that would depend on the source file and then make the test target depend on this additional target (seems a bit hacky btw, but I guess cmake comes with some limitations). Unfortunately, you cannot tell `yoda_add_unittest` to depend on another target unless it's a linkable library (`DEPENDS` keyword).
Another thing that I noticed is that probably the `.hpp` extension was assigned to these files (which come from the prototype) because they contain a stencil class which is templated (with respect to the library tag) ... don't know if this affects something here, though.